pkg/cache/cluster.go

```diff
@@ -813,6 +815,53 @@ func (c *clusterCache) IterateHierarchy(key kube.ResourceKey, action func(resour
 	}
 }
 
+// IterateGroupHierarchy executes multiple IterateHierarchy operations and only takes lock once.
+func (c *clusterCache) IterateGroupHierarchy(keys []kube.ResourceKey, action func(resource *Resource, namespaceResources map[kube.ResourceKey]*Resource)) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	var wg sync.WaitGroup
+	for _, key := range keys {
```

Comment:
Instead of copying whole `IterateHierarchy` method it is better to create private `iterateHierarchy` method that does not use lock and use it in both `IterateGroupHierarchy` and `IterateHierarchy`